sub-packages/bionemo-example_model/README.md

```diff
@@ -1,7 +1,115 @@
+
+
+We also define a train_dataloader, val_dataloader, and predict_dataloader methods that return the corresponding dataloaders.
+
+# Models
+
+Models need to be Megatron modules. At the most basic level this just means:
+
+1. They extend `MegatronModule` from megatron.core.transformer.module.
+2. They need a config argument of type `megatron.core.ModelParallelConfig`. An easy way of implementing this is to inherit from `bionemo.llm.model.config.MegatronBioNeMoTrainableModelConfig`. This is a class for BioNeMo that supports usage with Megatron models, as NeMo2 requires. This class also inherits `ModelParallelConfig`.
+3. They need a self.`model_type:megatron.core.transformer.enums.ModelType` enum defined (`ModelType.encoder_or_decoder` is usually fine)
+4. def set_input_tensor(self, input_tensor) needs to be present. This is used in model parallelism. This function can be a stub/placeholder function.
+
+`ExampleModelTrunk` is a base model. This returns a tensor. `ExampleModel` is a model that extends the base model with a few linear layers and it is used for pretrainining. This returns the output of the base model and of the full model.
+
+`ExampleFineTuneModel` extends the `ExampleModelTrunk` by adding a classification layer. This returns a tensor of logits over the 10 potential digits.
+
+# Model Configs
+
+The model config class is used to instantiate the model. These configs must have:
+1. A configure_model method which allows the Megatron strategy to lazily initialize the model after the parallel computing environment has been setup. These also handle loading starting weights for fine-tuning cases. Additionally these configs tell the trainer which loss you want to use with a matched model.
+2. A get_loss_reduction_class method that defines the loss function.
+
+Here, a base generic config `ExampleGenericConfig` is defined.  `PretrainConfig` extends this class. This defines the model class and the loss class in:
 ```
+class PretrainConfig(ExampleGenericConfig["PretrainModel", "MSELossReduction"], iom.IOMixinWithGettersSetters):
+
+    model_cls: Type[PretrainModel] = PretrainModel
+    loss_cls: Type[MSELossReduction] = MSELossReduction
+
+```
+
+Similarly, `ExampleFineTuneConfig` extends `ExampleGenericConfig` for finetuning.
+
+# Training Module
+
+It is helfpul to have a training module that inherits from `pl.LightningModule` which organizes the model architecture, training, validation, and testing logic while abstracting away boilerplate code, enabling easier and more scalable training. This wrapper can be used for all model and loss combinations specified in the config.
```

Comment:
is `pl` namespace here consistent? we use `lightning` elsewhere in these docs.

Reply:
There's also `nl`??

Reply:
pl = pytorch_lightning
nl = nemo.lightning. Will make this clear.